src/cmd_sym.ml

```diff
@@ -168,24 +168,24 @@ let simplify_then_link_then_run ~unsafe ~optimize (pc : unit Result.t Choice.t)
   in
   List.fold_left f pc (List.rev to_run)
 
-let run_file ~unsafe ~optimize (pc : unit Result.t Choice.t) filename =
-  if not @@ Sys.file_exists filename then
-    Choice.return (error_s "file `%s` doesn't exist" filename)
-  else
-    let*/ script = Parse.Script.from_file ~filename in
-    simplify_then_link_then_run ~unsafe ~optimize pc script
+let run_file ~unsafe ~optimize pc filename =
+  let*/ script = Parse.Script.from_file filename in
+  simplify_then_link_then_run ~unsafe ~optimize pc script
 
 let get_model solver pc =
   assert (Solver.check solver pc);
   match Solver.model solver with None -> assert false | Some model -> model
 
-let mkdir_p_exn dir =
-  let rec get_intermediate_dirs d acc =
-    if Sys.file_exists d then acc
-    else get_intermediate_dirs (Filename.dirname d) (d :: acc)
+let mkdir_p_exn (dir : Fpath.t) =
+  let rec get_intermediate_dirs (d : Fpath.t) acc =
+    if Sys.file_exists (Fpath.to_string d) then acc
+    else
+      get_intermediate_dirs
+        (Fpath.normalize d |> Fpath.split_base |> fst)
+        (d :: acc)
   in
   let intermediate_dirs = get_intermediate_dirs dir [] in
-  List.iter (fun d -> Sys.mkdir d 0o755) intermediate_dirs
+  List.iter (fun d -> Sys.mkdir (Fpath.to_string d) 0o755) intermediate_dirs
```

Comment:
I guess this function can be removed in favour of `OS.Dir.create ~path:true`

Reply:
Indeed. Done (with others along the way) :-)